omigami/tests/test_spectra_matching.py

```diff
@@ -0,0 +1,148 @@
+        "Precursor_MZ": "153.233",
+    }
+    client = SpectraMatching()
+    # first validates if the input is correct then we test for errors
+    client._validate_input([model_input])
+
+    with pytest.raises(TypeError, match="Spectrum data must be a dictionary."):
+        client._validate_input(["not_a_dict"])
+
+    with pytest.raises(KeyError, match="mandatory keys"):
+        client._validate_input([{"Precursor_MZ": "1", "peaks_JASON": "[not a list]"}])
+
+    with pytest.raises(
+        ValueError, match="peaks_json needs to be a valid python string representation"
+    ):
+        client._validate_input([{"Precursor_MZ": "1", "peaks_json": "[not a list]"}])
+
+    with pytest.raises(
+        ValueError,
+        match="peaks_json needs to be a valid python string representation",
+    ):
+        client._validate_input([{"Precursor_MZ": "1", "peaks_json": 10}])
+
+    with pytest.raises(
+        ValueError,
+        match="Precursor_MZ needs to be a string representation of a float",
+    ):
+        client._validate_input([{"Precursor_MZ": "float", "peaks_json": [10]}])
+
+
+def test_create_spectra_generator_file_source(small_mgf_path):
+    client = SpectraMatching()
+
+    spectra_generator = client._create_spectra_generator(small_mgf_path)
+
+    assert isinstance(spectra_generator.__next__(), Spectrum)
+
+
+def test_create_spectra_generator_stream_source(small_mgf_path):
+    client = SpectraMatching()
+    stream = StringIO(open(small_mgf_path, "r").read())
+
+    spectra_generator = client._create_spectra_generator(stream)
+
+    assert isinstance(spectra_generator.__next__(), Spectrum)
+
+
+def test_create_spectra_generator_spectrum_source(small_mgf_path):
+    client = SpectraMatching()
+    spectra = list(load_from_mgf(small_mgf_path))
+
+    spectra_generator = client._create_spectra_generator(spectra)
+
+    assert isinstance(spectra_generator.__next__(), Spectrum)
+
+
+def test_make_batch_requests(small_mgf_path, sample_response):
+    client = SpectraMatching()
+    _46_spectra = list(load_from_mgf(small_mgf_path))
+
+    def _send_request(payload: Payload, *args) -> list:
+        """Creates a dummy response of the same size of the payload"""
+        payload_size = len(payload["data"]["ndarray"]["data"])
+        return payload_size * ["result"]
+
+    client._send_request = Mock(side_effect=_send_request)
+    client._format_results = lambda r: r  # to not do any formatting and return as it is
+
+    # Input has 260 spectra -> 3 batch requests of [100, 100, 60]
+    input_spectra = _46_spectra * 5
+    n_spectra = len(input_spectra)
+    spectra_generator = client._create_spectra_generator(input_spectra)
+
+    predictions = client._make_batch_requests(
+        spectra_generator, {"n_best_spectra": 2}, "endpoint"
+    )
+
+    assert len(predictions) == n_spectra
+    assert client._send_request.call_count == n_spectra // SPECTRA_LIMIT_PER_REQUEST + 1
```

Comment:
since this test name is pretty general "test_make_batch_requests" I would like a short docstring explaining what this tests specifically (That it tests if the function is doing the correct amount of requests (_send_request calls) considering size of input and limit of spectra per request)

btw, is the first assertion necessary? Since we are faking the response (predictions) isn't it a given that this will pass? Or I didn't get it?

Reply:
The mocking of `send_request` just asserts that if you query for 30 spectra, you will get 30 spectra in your response - it mirrors the behaviour of a working endpoint. 

The first assertion is about being sure all spectra from the user input are batched correctly and included in the requests. It assures that the batching is working fine. We could still have a passing call_count assertion and for some reason not all spectra were being included

Reply:
> "it tests if the function is doing the correct amount of requests". 

This is the batching logic itself. The behaviour of the batching logic is being tested. There is no other thing to test in this code IMO.